pkg/graveler/committed/merge.go

```diff
@@ -370,7 +370,8 @@ func (m *merger) handleDestRangeSourceKey(destRange *Range, sourceValue *gravele
 		return m.sourceBeforeDest(sourceValue)
 	}
 
-	if bytes.Compare(destRange.MaxKey, sourceValue.Key) < 0 { // dest range before source
+	if bytes.Compare(destRange.MaxKey, sourceValue.Key) < 0 &&
+		(m.strategy != graveler.MergeStrategyImport || m.validImportRange()) { // dest range before source
```

Comment:
graveler merge code should not address special import rage by casting iterator implementation. the solution should be a general solution that works with any merge or by provide merge options that will customize the merge we try to apply, but by graveler merge knowing this is merge.